File: functions/model/FortuneUser/components/NotificationSettings.js
class NotificationSettings {
    constructor({ surpriseNotification, fortuneReadyNotification, dailyNews }) {
        this.surpriseNotification = surpriseNotification;
        this.fortuneReadyNotification = fortuneReadyNotification;
        this.dailyNews = dailyNews;


    }

    static fromJson(json) {
        return new NotificationSettings({
            surpriseNotification: json["surpriseNotification"],
            fortuneReadyNotification: json["fortuneReadyNotification"],
            dailyNews: json["dailyNews"],
        });
    }

    toJson() {
        return {
            "surpriseNotification": this.surpriseNotification,
            "fortuneReadyNotification": this.fortuneReadyNotification,
            "dailyNews": this.dailyNews,
        };
    }
}
module.exports = { NotificationSettings };

File: lib/src/model/FortuneUser/components/notification_settings.dart

class NotificationSettings {
  bool surpriseNotification;
  bool fortuneReadyNotification;
  bool dailyNews;

  NotificationSettings({
    required this.surpriseNotification,
    required this.fortuneReadyNotification,
    required this.dailyNews,
  });

  factory NotificationSettings.fromJson(Map<String, dynamic> json) => NotificationSettings(
        surpriseNotification: json["surpriseNotification"],
        fortuneReadyNotification: json["fortuneReadyNotification"],
        dailyNews: json["dailyNews"],
      );

  Map<String, dynamic> toJson() => {
        "surpriseNotification": surpriseNotification,
        "fortuneReadyNotification": fortuneReadyNotification,
        "dailyNews": dailyNews,
      };
}